activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -37,15 +37,26 @@ def explain(arel, binds = [])
           MySQL::ExplainPrettyPrinter.new.pp(result, elapsed)
         end
 
+        def execute(sql, name = nil, async: false)
```

Comment:
A class defining the method twice is very strange. Why is this not inside the second `execute`?

Reply:
If we need a `raw_execute` I think it is better to define it in the abstract adapter and in all other than doing this trick. It is hard to understand why it is needed here.

Reply:
Changed in https://github.com/rails/rails/commit/55199212542d8d8536e5bb93c43f1be50730ae74

Reply:
👍